loader/include/Geode/ui/ProgressBar.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <cocos2d.h>
+
+namespace geode {
+    // Custom class for the progress bar
+    class ProgressBar : public cocos2d::CCNode {
+    protected:
+        Ref<CCSprite> m_progressBar = nullptr; // Progress bar outline
+        CCSprite* m_progressBarFill = nullptr; // Progress bar fill
+
+        float m_progress = 0.f; // Current progress bar fill percentage ranging from 0 to 100
+
+        float m_progressBarFillMaxWidth = 0.f; // Max width for the progress fill bar node
+        float m_progressBarFillMaxHeight = 0.f; // Max height for the progress fill bar node
```

Comment:
Class members should be PIMPL; aka these should be behind a source-defined `Impl` class and `std::unique_ptr<Impl> m_impl`. See other Geode classes for examples